### ISSUE TITLE
Example in `profiles.yml` for `indirect_selection`

### ISSUE THREAD

**Issue by dbeatty10** [fishtown-analytics/docs.getdbt.com#4773]
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Two pages:
- https://docs.getdbt.com/docs/core/connect-data-platform/profiles.yml
- new page named something like https://docs.getdbt.com/reference/global-configs/indirect-selection or https://docs.getdbt.com/reference/global-configs/selection, etc.

The new page should work in concert with these pre-existing page sections:
- https://docs.getdbt.com/reference/node-selection/test-selection-examples#indirect-selection
- https://docs.getdbt.com/reference/node-selection/test-selection-examples#syntax-examples
- https://docs.getdbt.com/reference/node-selection/yaml-selectors#indirect-selection

### What part(s) of the page would you like to see updated?

Starting in v1.8, https://github.com/dbt-labs/dbt-core/pull/9040 adds `indirect_selection` as a global config, so we should:

1. create a global config page for this flag
2. update [`profiles.yml`](https://docs.getdbt.com/docs/core/connect-data-platform/profiles.yml) accordingly

If we were to _just_ do number 2) without doing 1), then the update might look like this (with applicable version blocks, of course):
```
  [indirect_selection](/reference/node-selection/test-selection-examples#indirect-selection): <eager | cautious | buildable | empty>
```

**Comment by mirnawong1** [comment]
hey @dbeatty10 is this issue for dbt v1.8?

**Comment by dbeatty10** [comment]
> hey @dbeatty10 is this issue for dbt v1.8?

It is!

Just added a `dbt-core 1.8` label. Is there anything else we should do given that it's for v1.8?

**Comment by runleonarun** [comment]
### @runleonarun adding copilot summary

[This pull request](https://github.com/dbt-labs/dbt-core/pull/9040) primarily focuses on consolidating deferral methods and flags within the `dbt` project. The changes involve modifications to various methods and classes across multiple files, including `core/dbt/cli/flags.py` and `core/dbt/cli/main.py`. The most significant changes include the addition of a new `params_assigned_from_user` set and changes to the `_assign_params` method, reordering and removal of decorators in `core/dbt/cli/main.py`, and the introduction of `defer_relation` in `core/dbt/contracts/graph/manifest.py`.

<details>

1. **Changes to `_assign_params` method in `core/dbt/cli/flags.py`**:
    * A new set `params_assigned_from_user` is introduced and used to track user-provided and default values across both 'parent' and 'child' levels. This is to support detection of mutually exclusive flags later on. [[1]](diffhunk://#diff-3e5c2add494f2ffa25c42dd0ea6ea0d4cc43875474b110506dff44385adb81acR116) [[2]](diffhunk://#diff-3e5c2add494f2ffa25c42dd0ea6ea0d4cc43875474b110506dff44385adb81acL181-R205)
    * The `_assign_params` method is modified to include `params_assigned_from_user` as a parameter. [[1]](diffhunk://#diff-3e5c2add494f2ffa25c42dd0ea6ea0d4cc43875474b110506dff44385adb81acL181-R205) [[2]](diffhunk://#diff-3e5c2add494f2ffa25c42dd0ea6ea0d4cc43875474b110506dff44385adb81acL206-R225)

2. **Reordering and removal of decorators in `core/dbt/cli/main.py`**:
    * Several decorators are added to the `global_flags` function and removed from other functions. This change is likely aimed at consolidating the use of these decorators. [[1]](diffhunk://#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6R124-R136) [[2]](diffhunk://#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6R152-R160) [[3]](diffhunk://#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6L167-L169) and others)

3. **Introduction of `defer_relation` in `core/dbt/contracts/graph/manifest.py`**:
    * For non-ephemeral refable nodes, a new `defer_relation` attribute is added. This attribute is used when the model is deferred and the adapter doesn't support zero-copy cloning. [[1]](diffhunk://#diff-d14b7d00e52da5da96bfb3bdef414e9289d923efe0a9cdadd716d792c7199535L1409-R1410) [[2]](diffhunk://#diff-1e80746bd25a5ea5ca6604030a2991dfbae0e486f971a6665f2b3a3512f05532R1373-R1379)

4. **Changes in `core/dbt/task/clone.py`**:
    * The `CloneTask` class now always requires a state manifest, regardless of whether the `--defer` flag has been set.
    * The `defer_to_manifest` method is removed. Unlike other commands, 'clone' always requires a state manifest.

5. **Consolidation of deferral methods and flags**:
    * An 'Under the Hood' change is added to consolidate deferral methods and flags.

These changes seem to be part of a larger refactoring effort aimed at simplifying the codebase and improving the handling of deferral methods and flags.

</details>

**Comment by runleonarun** [comment]
Hey @dbeatty10 The [linked PR](https://github.com/dbt-labs/dbt-core/pull/9040) doesn't mention indirect_selection as a global config. Is there another PR we should be aware of for this issue? cc @graciegoheen

**Comment by dbeatty10** [comment]
> Hey @dbeatty10 The [linked PR](https://github.com/dbt-labs/dbt-core/pull/9040) doesn't mention indirect_selection as a global config. Is there another PR we should be aware of for this issue? cc @graciegoheen

@runleonarun This is indeed the correct PR!

`indirect_selection` was removed from individual sub commands and [added to `global_flags`](https://github.com/dbt-labs/dbt-core/blob/2411f93240a346961e6a965ad9cb5c766db275a0/core/dbt/cli/main.py#L120-L133) (which is included in all sub commands) within [#9040](https://github.com/dbt-labs/dbt-core/pull/9040/files).

**Comment by runleonarun** [comment]
Closed in https://github.com/dbt-labs/docs.getdbt.com/pull/4977
